.github/workflows/app.yml

```diff
@@ -0,0 +1,48 @@
+name: app
+on:
+  - merge_group
+  - push
+  - pull_request
+
+# Only build for latest push/PR unless it's main or release/
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' && !startsWith( github.ref, 'refs/heads/release/' ) && !startsWith( github.ref, 'refs/heads/gh-readonly-queue/' ) }}
+
+defaults:
+  run:
+    shell: bash
+
+permissions: {}
+
+jobs:
+  lint-and-build:
+    strategy:
+      matrix:
+        debian_version:
+          - bookworm
+    runs-on: ubuntu-latest
+    container: debian:${{ matrix.debian_version }}
+    steps:
+      - name: Install base dependencies
+        run: |
+          apt-get update && apt-get install --yes git make
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+      - name: Install node
+        uses: actions/setup-node@v4
+        with:
+          node-version: "22.x"
+          cache: npm
+          cache-dependency-path: app/package-lock.json
+      - name: Install npm dependencies
+        run: |
+          cd app; npm ci
+      - name: Lint
+        run: |
+          cd app; npm run lint
+      - name: Build
+        run: |
+          cd app; npm run package
```

Comment:
Nit:

```suggestion
      - name: Install npm dependencies
        run: |
          npm ci
        working-directory: app
      - name: Lint
        run: |
          npm run lint
        working-directory: app
      - name: Build
        run: |
          npm run package
        working-directory: app
```

Reply:
Done. Note that at one point when making this PR I considered setting up a Makefile wrapper so we could do e.g. `make -C lint` but then felt it was a bit silly since it would just be a passthrough to npm, which has a decent task runner already. And we can always add a Makefile later if the muscle-memory/consistency would be useful.